File: Assets/VirtualCity/ProtoDefine/Java/building/ReqOpenAreaMessage.java
package com.kingston.jforgame.server.game.building.message;

import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;
import com.kingston.jforgame.server.game.Modules;
import com.kingston.jforgame.server.game.building.BuildingDataPool;
import com.kingston.jforgame.socket.annotation.MessageMeta;
import com.kingston.jforgame.socket.message.Message;

@MessageMeta(module = Modules.BULIDING,cmd = BuildingDataPool.REQ_OPENAREA)

public class ReqOpenAreaMessage extends Message {
    @Protobuf(order = 1)
    private String code;
    /**
     * 0:金币
     * 1:钻石
     */
    @Protobuf(order = 2)
    private int costDiamond;


    public ReqOpenAreaMessage() {
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public int getCostDiamond() {
        return costDiamond;
    }

    public void setCostDiamond(int costDiamond) {
        this.costDiamond = costDiamond;
    }
}

File: Assets/VirtualCity/ProtoDefine/ReqOpenAreaMessage.cs
using System.Collections.Generic;
using ProtoBuf;
namespace ProtoDefine {
[ProtoContract]



public class ReqOpenAreaMessage {
[ProtoMember(1)]
    public string code;
    /**
     * 0:金币
     * 1:钻石
     */
[ProtoMember(2)]
    public int costDiamond;


    public ReqOpenAreaMessage() {
    }

    public string getCode() {
        return code;
    }

    public void setCode(string code) {
        this.code = code;
    }

    public int getCostDiamond() {
        return costDiamond;
    }

    public void setCostDiamond(int costDiamond) {
        this.costDiamond = costDiamond;
    }
}
}